src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -37,17 +37,49 @@ define([
       rollBackApp: React.PropTypes.func.isRequired,
       scaleApp: React.PropTypes.func.isRequired,
       suspendApp: React.PropTypes.func.isRequired,
-      tasksFetchState: React.PropTypes.number.isRequired
+      tasksFetchState: React.PropTypes.number.isRequired,
+      router: React.PropTypes.object.isRequired
     },
 
     getInitialState: function() {
+      var appid = this.props.model.get('id');
+
+      this.tabs = [];
+
+      _.each(tabsTemplate, function(tab) {
```

Comment:
Yeah, this is a better initialization of this.tabs.
But you have to return current to set the memo..